proto/plan_common.proto

```diff
@@ -27,8 +27,16 @@ message ColumnDesc {
   // For example, when the type is created from a protobuf schema file,
   // this field will store the message name.
   string type_name = 5;
-  // Optional description for the generated column.
-  GeneratedColumnDesc generated_column = 6;
+
+  // deprecated
+  // GeneratedColumnDesc generated_column = 6 [deprecated = true];
+  reserved 6;
+
+  // Optional description for the generated column or default value.
+  oneof generated_or_default_column {
+    GeneratedColumnDesc generated_column = 7;
+    DefaultColumnDesc default_column = 8;
+  }
```

Comment:
https://protobuf.dev/programming-guides/proto3/#updating

> Changing a single optional field or extension into a member of a new oneof is binary compatible, however for some languages (notably, Go) the generated code’s API will change in incompatible ways. For this reason, Google does not make such changes in its public APIs, as documented in [AIP-180](https://google.aip.dev/180#moving-into-oneofs). With the same caveat about source-compatibility, moving multiple fields into a new oneof may be safe if you are sure that no code sets more than one at a time. Moving fields into an existing oneof is not safe. Likewise, changing a single field oneof to an optional field or extension is safe.

So I guess this is compatible as well. We can ignore the programming language API incompatibility.